app/components/app_consent_component.rb

```diff
@@ -55,20 +55,36 @@ def contact_parent_or_guardian_link(consents)
 
   def consents_grouped_by_parent
     @consents_grouped_by_parent ||=
-      @patient_session.consents.group_by { |consent| summary_for(consent) }
+      @patient_session.consents.group_by(&:summary_with_consenter)
   end
 
   private
 
-  def summary_for(consent)
-    summary =
-      if consent.respond_to?(:response_not_provided?) &&
-           consent.response_not_provided?
-        "Contacted #{consent.name}"
-      else
-        "#{consent.human_enum_name(:response).capitalize} by #{consent.name}"
-      end
-    summary += " (#{consent.who_responded})" unless consent.via_self_consent?
-    summary
+  def render_grouped_consents(consents)
+    first_consent = consents.first
+    first_refused_consent = consents.find_all(&:response_refused?).first
+
+    props = {
+      name: first_consent.name,
+      response:
```

Comment:
This is a list of responses, so plural?

```suggestion
      responses:
```

Reply:
nvm, I see below that it can be a single `response` in some circumstances too. 🤔

Reply:
I've left it as `response` for now, as that way it can be mandatory. I suppose there's an option to allow both `response` and `responses` but that would require checking that at least one is populated and picking the right one within the component 🤷 . I can do that if needed...